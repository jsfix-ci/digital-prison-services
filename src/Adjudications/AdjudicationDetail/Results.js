import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { H3 } from '@govuk-react/heading/lib/presets'
import { LabelAndValue } from './AdjudicationDetail.styles'

export const Results = ({ results }) => (
  <>
    <H3>Results</H3>
    {results && results.length ? (
      results.map(result => (
        <Fragment key={result.id}>
          <GridRow>
            <GridCol>
              <LabelAndValue label="Offence Paragraph" value={result.oicOffenceCode} />
            </GridCol>
            <GridCol>
              <LabelAndValue label="Offence Type" value={result.offenceType} />
            </GridCol>
            <GridCol>
              <LabelAndValue label="Offence Description" value={result.offenceDescription} />
            </GridCol>
            <GridCol>
              <LabelAndValue label="Plea" value={result.plea} />
            </GridCol>
            <GridCol>
              <LabelAndValue label="Finding" value={result.finding} />
            </GridCol>
          </GridRow>
        </Fragment>
      ))
    ) : (
      <p>No information available</p>
    )}
    <hr />
  </>
)

Results.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      oicOffenceCode: PropTypes.string,
      offenceType: PropTypes.string,
      offenceDescription: PropTypes.string,
      plea: PropTypes.string,
      finding: PropTypes.string,
    })
  ),
}

const mapStateToProps = ({
  adjudicationHistory: {
    detail: { results },
  },
}) => ({
  results,
})

export default connect(mapStateToProps)(Results)
