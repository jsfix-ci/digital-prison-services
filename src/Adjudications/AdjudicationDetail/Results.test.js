import React from 'react'
import testRenderer from 'react-test-renderer'

import { Results } from './Results'

describe('Results', () => {
  it('should render no results correctly', () => {
    const results = []

    const wrapper = testRenderer.create(<Results results={results} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render present results correctly', () => {
    const results = [
      {
        id: '1',
        oicOffenceCode: '12:21',
        offenceType: 'Adult',
        offenceDescription: 'Assault',
        plea: 'Not Guilty',
        finding: 'Guilty',
      },
      {
        id: '2',
        oicOffenceCode: '23D',
        offenceType: 'Adult',
        offenceDescription: 'Stealing',
        plea: 'Not Guilty',
        finding: 'Not Proved',
      },
    ]

    const wrapper = testRenderer.create(<Results results={results} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
