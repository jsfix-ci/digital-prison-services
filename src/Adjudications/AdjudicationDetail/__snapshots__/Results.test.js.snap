// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render no results correctly 1`] = `
Array [
  <h3
    className="sc-htpNat jszPot"
  >
    Results
  </h3>,
  <p>
    No Information Available
  </p>,
  <hr />,
]
`;

exports[`Results should render present results correctly 1`] = `
Array [
  <h3
    className="sc-htpNat jszPot"
  >
    Results
  </h3>,
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Paragraph
      </strong>
      <p>
        12:21
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Type
      </strong>
      <p>
        Adult
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Description
      </strong>
      <p>
        Assault
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Plea
      </strong>
      <p>
        Not Guilty
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Finding
      </strong>
      <p>
        Guilty
      </p>
    </div>
  </div>,
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Paragraph
      </strong>
      <p>
        23D
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Type
      </strong>
      <p>
        Adult
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Offence Description
      </strong>
      <p>
        Stealing
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Plea
      </strong>
      <p>
        Not Guilty
      </p>
    </div>
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Finding
      </strong>
      <p>
        Not Proved
      </p>
    </div>
  </div>,
  <hr />,
]
`;
